### PR TITLE
Finishing Move store credits in core

### PR DESCRIPTION
**Pull request by peterberkenbosch**
branched from this PR #5625, rebased on master. This is a WIP to finish the move.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://peterberkenbosch-5625-store-credits.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Click to open.</a>

File: core/lib/generators/spree/install/install_generator.rb [hunk 1]
```diff
@@ -111,6 +111,7 @@ def include_seed_data
 \n
 Spree::Core::Engine.load_seed if defined?(Spree::Core)
 Spree::Auth::Engine.load_seed if defined?(Spree::Auth)
+SpreeStoreCredits::Engine.load_seed if defined?(SpreeStoreCredits::Engine)
```
Comment (JDutil) [inline core/lib/generators/spree/install/install_generator.rb]: seeds should be part of core not changing this generator, and same for the assets below

File: core/db/migrate/20140523190528_create_spree_store_credit_tables.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    add_index :spree_store_credit_events, :store_credit_id
+
+    create_table :spree_store_credit_types do |t|
+      t.string :name
+      t.integer :priority
+      t.timestamps
+    end
+    add_index :spree_store_credit_types, :priority
+
+    default_type = Spree::StoreCreditType.find_or_create_by(name: 'Expiring', priority: 1)
```
Comment (JDutil) [inline core/db/migrate/20140523190528_create_spree_store_credit_tables.rb]: I don't think we should be seeding any of this data

File: core/config/routes.rb [hunk 3]
```diff
@@ -1 +1,17 @@
...
+Spree::Core::Engine.draw_routes
+
+Spree::Core::Engine.routes.draw do
+  namespace :admin do
+    resources :users, only: [] do
+      resources :store_credits
+    end
+  end
+
+  namespace :api, defaults: { format: 'json' } do
```
Comment (JDutil) [inline core/config/routes.rb]: belongs in api/config/routes

File: core/config/routes.rb [hunk 4]
```diff
@@ -1 +1,17 @@
-Spree::Core::Engine.draw_routes
\ No newline at end of file
+Spree::Core::Engine.draw_routes
+
+Spree::Core::Engine.routes.draw do
+  namespace :admin do
```
Comment (JDutil) [inline core/config/routes.rb]: belongs in backend/config/routes

File: core/app/models/spree/payment.rb [hunk 5]
```diff
@@ -222,5 +229,42 @@ def update_order
         end
       end
 
+      # Necessary because some payment gateways will refuse payments with
```
Comment (JDutil) [inline core/app/models/spree/payment.rb]: I think all of this except for create_eligible_credit_event method should be removed

File: core/app/models/spree/order/checkout.rb [hunk 6]
```diff
@@ -102,6 +102,8 @@ def self.define_state_machine!
               before_transition to: :resumed, do: :ensure_line_item_variants_are_not_deleted
               before_transition to: :resumed, do: :ensure_line_items_are_in_stock
 
+              before_transition to: :confirm, do: :add_store_credit_payments
```
Comment (JDutil) [inline core/app/models/spree/order/checkout.rb]: can't rely on confirm step existing for non payment profile gateways

Comment (paultyng) [inline core/app/models/spree/order/checkout.rb]: See [my comment](https://github.com/spree/spree/pull/5625#issuecomment-69361879) in #5625  for how we handled it on the existing extension.

**Comment by peterberkenbosch** [comment]
I will take another shot at this with a clean state, based on the bonobos implementation. Can't verify the state this is in right now. Closing this now and will open a new one.

**Pull request closed without merging**